### DISCUSSION TITLE
Enhancing Copilot usage API with repo URLs

### DISCUSSION THREAD

**Discussion by ghfrontierman** [community/community#112377] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

Hello GitHub Team and Community, I have started to evaluate the [GitHub Copilot usage REST API](https://docs.github.com/en/early-access/copilot/copilot-usage-api) available under early access.

After gaining experience with this API over an extended period, I began wondering how to correlate individual Git repos with the usage data points that this API returns.

This got me wondering if the breakdown structure could be shaped to include the 'repository' as part of the json response. Possibly a response that might include the git remote URL, something like this example:

  "breakdown": [
      {
        "repository": "https://github.com/myproject/mysvc.git",
        "language": "python",
        "editor": "vscode",
        "suggestions_count": 300,
        "acceptances_count": 200,
        "lines_suggested": 600,
        "lines_accepted": 300,
        "active_users": 3
      },
     ..
     ..

Including the 'repository' field would further enhance the utility of the data by allowing users to correlate Copilot's usage metrics with specific Git repos directly.

I would like to hear feedback from GitHub and the community on this suggestion and whether it aligns with the vision for the Copilot usage API. Thank you for considering my feedback.

**Comment by riverjang-nsuslab** [comment]
I think this is the minimum amount of information that we need for Copilot enterprise/organization. 
We need to assign usage to each project / repository by looking at usage per repository level at the very least. 

I think we will eventually need to have information regarding individual user usage, but I agree that usage per repository is the first step.

**Comment by norbertfijaleklingaro** [comment]
Hello, fully agreed! We are also exploring the possibility of **measuring GitHub Copilot usage on a per-repository or group-of-repositories** basis, as the current API only offers usage metrics per individual or team. 
GitHub Team, could you extend the GitHub Copilot API to include this functionality(stats per repository/group of repositories) or suggest a workaround or alternative approach to access such data? 
**This capability is crucial for us to monitor adoption progress for one of our global FMCG clients, and this request is quite urgent for us, and we’d love to get to know do we have some possibilities of cooperation here.**
